### ISSUE TITLE
Cannot have trailing commas in tuples in generic parameters

### ISSUE THREAD

**Issue by simonwelsh** [facebook/hhvm#6199]
``` hack
<?hh

function foo(): ConstVector<(
    string,
    int,
)> {}
```

gives `Fatal error: syntax error, unexpected '<', expecting ';' or '{' in /in/UvWrJ on line 3`

However, both:

``` hack
<?hh

function foo(): (
    string,
    int,
) {}
```

(just a tuple, with trailing comma) and

``` hack
<?hh

function foo(): ConstVector<(
    string,
    int
)> {}
```

(tuple inside generics, without a trailing comma) are valid.

**Comment by simonwelsh** [comment]
(I feel like this is a bug with HHVM, rather than `hh_format`)

**Comment by fredemmott** [comment]
Yep, but given we've just shipped 3.9, should probably change hh_format too (temporarily). Don't want to say libraries can't use hh_format if they care about LTS releases

--
Typed with my thumbs

On Sep 4, 2015, at 18:41, Simon Welsh <notifications@github.com<mailto:notifications@github.com>> wrote:


(I feel like this is a bug with HHVM, rather than hh_format)

**Comment by simonwelsh** [comment]
Has been fixed
